src/main/java/gov/cabinetofice/gapuserservice/web/SuperAdminController.java

```diff
@@ -27,22 +35,65 @@ public class SuperAdminController {
+    @GetMapping("/super-admin-dashboard") ResponseEntity<?> superAdminDashboard(
+            final HttpServletRequest httpRequest,
+            final Pageable pagination,
+            @RequestParam(value = "departments", required = false) String departments,
+            @RequestParam(value = "roles", required = false) String roles,
+            @RequestParam(value = "searchTerm", required = false) String searchTerm,
+            @RequestParam(value = "clearAllFilters", required = false) boolean clearAllFilters
+    ){
         if (!roleService.isSuperAdmin(httpRequest)) {
             throw new ForbiddenException();
         }
 
-        List<DepartmentDto> departments = departmentService.getAllDepartments();
-        List<RoleDto> roles = roleService.getAllRoles();
-        List<UserDto> users = oneLoginUserService.getPaginatedUsers(pagination);
+        if(searchTerm != null){
+            if(searchTerm.length() > 255){
+                Error errorResponse = Error.builder().errorMessage("Search term must be less than 255 characters").fieldName("searchTerm").build();
+                ErrorResponseBody errorResponseBody = ErrorResponseBody.builder().responseAccepted(false).message("Search term must be less than 255 characters").errors(List.of(errorResponse)).build();
+                return new ResponseEntity(errorResponseBody, HttpStatus.BAD_REQUEST);
+             }
+        }
+
+        List<DepartmentDto> allDepartments = departmentService.getAllDepartments();
+        List<RoleDto> allRoles = roleService.getAllRoles();
+        List<UserDto> users = oneLoginUserService.getPaginatedUsers(pagination, searchTerm);
         long userCount = oneLoginUserService.getUserCount();
-        
+
+        if (clearAllFilters || (departments == null && roles == null)){
+            return ResponseEntity.ok(SuperAdminDashboardPageDto.builder()
+                    .departments(allDepartments)
+                    .roles(allRoles)
+                    .users(users)
+                    .userCount(userCount)
+                    .build());
+        }
+
+        List<String> departmentsList = Arrays.stream(departments.split(",")).toList();
+        List<String> rolesList = Arrays.stream(roles.split(",")).toList();
+
+        if(departments == "") {
+            departmentsList = allDepartments.stream().map(DepartmentDto::getId).collect(Collectors.toList());
+
+        }
+        if(roles == "") {
+            rolesList = allRoles.stream().map(RoleDto::getId).collect(Collectors.toList());
+        }
+
+        List<String> finalDepartments = departmentsList;
+        List<String> finalRoles = rolesList;
+        List<UserDto> filteredUsers = users.stream()
+                .filter(user -> finalDepartments.contains(String.valueOf(user.getDepartment().getId())))
+                .filter(user -> user.getRoles().stream().anyMatch(role -> finalRoles.contains(String.valueOf(role.getId()))))
+                .collect(Collectors.toList());
```

Comment:
won't this ruin pagination logic? Ideally this filtering should also be done in the DB query